runtimes/runtimes/operational-telemetry/aws-metric-exporter.ts

```diff
@@ -0,0 +1,172 @@
+                IdentityId: identityResponse.IdentityId,
+            })
+        )
+
+        if (!credentialsResponse.Credentials) {
+            throw new Error('No credentials received')
+        }
+
+        this.credentialsLastFetched = new Date()
+        this.credentials = credentialsResponse.Credentials
+
+        return this.credentials
+    }
+
+    signRequest(url: URL, body: string, region: string, credentials: Credentials) {
+        const request = new HttpRequest({
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                host: url.host,
+            },
+            hostname: url.host,
+            path: url.pathname,
+            body: body,
+        })
+
+        const signer = new SignatureV4({
+            credentials: {
+                accessKeyId: credentials.AccessKeyId!,
+                secretAccessKey: credentials.SecretKey!,
+                sessionToken: credentials.SessionToken!,
+            },
+            region: region,
+            service: 'execute-api',
+            sha256: Sha256,
+        })
+
+        return signer.sign(request)
+    }
+
+    extractOperationalMetrics(metrics: ResourceMetrics): OperationalMetric[] {
+        return metrics.scopeMetrics
+            .map((scopeMetrics: ScopeMetrics) => {
+                return scopeMetrics.metrics.map((metric: MetricData) => {
+                    return this.transform(metric, metrics.resource)
+                })
+            })
+            .flat()
+    }
+
+    transform(metric: MetricData, resource: Resource): OperationalMetric {
+        const dataPoint = metric.dataPoints[0]
+        return {
+            name: metric.descriptor.name,
+            value: dataPoint.value as number,
+            timestamp: dataPoint.endTime[0],
+            metrics: {},
+            atrributes: [],
+            server: {
+                name: resource.attributes['service.name'] as string,
+                version: resource.attributes['service.version'] as string,
+            },
+            clientInfo: {
+                name: this.telemetryService.getResource()['clientInfo.name'] as string,
+            },
+        }
+    }
+
+    async sendOperationalMetrics(metrics: OperationalMetric[]) {
+        for (const metric of metrics) {
+            const body = JSON.stringify(metric)
+            const url = new URL(this.endpoint)
+            const signedRequest = await this.signRequest(url, body, this.region, this.credentials!)
+
+            try {
+                const response = await axios({
+                    method: 'POST',
+                    url: this.endpoint,
+                    data: body,
+                    headers: signedRequest.headers,
+                })
+
+                // todo proper logging
+                console.log('Status Code:', response.status)
+                console.log('Response:', response.data)
+            } catch (e) {
```

Comment:
Retries for some cases may be needed. How will be know about these failures?

Reply:
We will log failures on in the standalone runtime, and we will also see requests on the api gateway side.